tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,168 @@
+import copy
+import pytest
+import re
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+OPENSSL_RECORD_WRITTEN_PATTERN = r"write to .*?\\n(.*?)\\n"
+OPENSSL_APP_DATA_HEADER_PATTERN = r"17 03 03 ([0-9a-f]{2} [0-9a-f]{2})"
+RECORD_SIZE_GROUP = 1
+
+
+def get_payload_size_from_openssl_trace(record_size_bytes: str) -> int:
+    # record_size_bytes is in the form XX XX where X is a hex digit
+    size_in_hex = record_size_bytes.replace(' ', '')
+    size = int(size_in_hex, 16)
+    # record includes 16 bytes of aead tag
+    return size - 16
+
+
+def assert_openssl_records_are_padded_correctly(openssl_output: str, padding_size: int):
+    number_of_app_data_records = 0
+
+    records_written = re.findall(
+        OPENSSL_RECORD_WRITTEN_PATTERN, openssl_output)
+    for record_prefix in records_written:
+        app_data_header = re.search(
+            OPENSSL_APP_DATA_HEADER_PATTERN, record_prefix)
+        if app_data_header:
+            size_bytes = app_data_header.group(RECORD_SIZE_GROUP)
+            size = get_payload_size_from_openssl_trace(size_bytes)
+
+            assert size > 0
+            assert size % padding_size == 0
+
+            number_of_app_data_records += 1
+
+    # At least one application record is sent + one close_notify alert. If openssl is
+    # the server then we also expect additional wrapped handshake records. Instead
+    # we assert the lower bound, which would be when openssl is the client.
+    assert number_of_app_data_records >= 2
```

Comment:
Both sides send encrypted handshake messages, not just the server. If you're not always seeing at least 3 app data records, it's possible the close_notify isn't always sent by both sides bc s2nc and s2nd are not very good at shutdowns. If that's the case, you can put it back to 2 and probably just not mention the close_notify in the comment.

```suggestion
    # The client and server write a variable number of encrypted handshake records,
    # but each write at least one (Finished). We also send at least one ApplicationData record,
    # and one final Alert record to end the connection.
    assert number_of_app_data_records >= 3
```